app/models/User.rb

```diff
@@ -17,4 +18,17 @@ def password=(password)
     @password = password
     self.password_hash = BCrypt::Password.create(password)
   end
+
+  def self.authenticate(email,password)
+  user = first(email: email)
+    if user 
```

Comment:
Kept nested if-else structure for granularity. Refactored abstract return values (`0` and `-1`) for clarity (`:incorrect_password` and `:nonexistent_user`).